### PR TITLE
Use structlog for logging

### PR DESCRIPTION
**Pull request by davidtsuk**
This PR transitions logging to use structlog as it has better compatibility with GCP. Currently, this uses a stdlib logging wrapper.

### CODE REVIEW

File: arroyo/backends/abstract.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+import structlog
 from abc import ABC, abstractmethod, abstractproperty
 from concurrent.futures import Future
 from typing import Callable, Generic, Mapping, Optional, Sequence, Union
+from arroyo.environment import setup_logging
 
 from arroyo.types import BrokerValue, Partition, Topic, TStrategyPayload
 
-logger = logging.getLogger(__name__)
+setup_logging()
```
Comment (evanh) [inline arroyo/backends/abstract.py:11]: If this call is always required before calling `structlog.get_logger` then I would wrap it in a single helper function.

```
def get_logger(name):
    setup_logging()
    return structlog.get_logger().bind(module=name)
```

File: arroyo/environment.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    """
+    Set the severity attribute for Google Cloud logging ingestion
+    """
+    event_dict["severity"] = event_dict["level"].upper()
+    del event_dict["level"]
+    return event_dict
+
+
+def setup_logging() -> None:
+    structlog.configure(
```
Comment (lynnagara) [inline arroyo/environment.py:19]: Shouldn't this get called once, rather than at the top of every module?

File: arroyo/environment.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        wrapper_class=structlog.stdlib.BoundLogger,
+        logger_factory=structlog.stdlib.LoggerFactory(),
+        processors=[
```
Comment (lynnagara) [inline arroyo/environment.py:22]: How does this work with custom processors, such as our Sentry integration? Doesn't this mean that users can't configure the logger?

**Review by lynnagara — COMMENTED** [review]
Not sure I get the motivation for this.

I know the the `Snuba` service is moving it's internal logging to use structlog, but I'm not sure if it's reasonable to expect all the open source libraries that are used by Snuba to also transition onto structlog.

Since the configure() call now gets made from every module of this library, how would the user configure the logger?

**Comment by lynnagara** [comment]
@davidtsuk Any update on this or can it be closed?

**Comment by davidtsuk** [comment]
I'm gonna close it

**Pull request closed without merging**